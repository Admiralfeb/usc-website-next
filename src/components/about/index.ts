export * from './allies';
export * from './faction';
export * from './hc';
export * from './rules';
export * from './carriers';
