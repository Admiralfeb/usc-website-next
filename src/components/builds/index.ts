export * from './builds/buildAdd';
export * from './builds/buildDetail';
export * from './queryandBuildList';
