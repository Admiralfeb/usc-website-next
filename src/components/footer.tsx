import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';

function ImgCopyright() {
  return (
    <Typography
      variant='body2'
      color='textSecondary'
      style={{ textAlign: 'center' }}>
      Thank you to CMDR RaZ uryel for the background image
    </Typography>
  );
}
function Copyright() {
  return (
    <Typography
      variant='body2'
      color='textSecondary'
      style={{ textAlign: 'center' }}>
      {'Copyright Â© '}
      <Link color='inherit' href='https://unitedsystems.org/'>
        United Systems Cooperative
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  footer: {
    padding: theme.spacing(3, 2),
    marginTop: 'auto',
    backgroundColor:
      theme.palette.type === 'light'
        ? theme.palette.grey[200]
        : theme.palette.grey[800],
  },
  center: {
    textAlign: 'center',
  },
}));

export const Footer = () => {
  const classes = useStyles();

  return (
    <footer className={classes.footer}>
      <Container maxWidth='sm' className={classes.center}>
        <Copyright />
        <ImgCopyright />
      </Container>
    </footer>
  );
};
