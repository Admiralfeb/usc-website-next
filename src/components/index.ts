export * from './home';
export * from './footer';
export * from './notFound';
export * from './auth/login';
