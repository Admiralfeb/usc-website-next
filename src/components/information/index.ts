export * from './infoSection';
export * from './infographic';
export * from './information';
export * from './miningMap';
export * from './shipReviews';
