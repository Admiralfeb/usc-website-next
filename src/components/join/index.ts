export * from './join';
export * from './joinList';
export * from './joinNew';
export * from './joinFormMember';
