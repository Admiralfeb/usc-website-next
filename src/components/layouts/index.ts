export * from './about';
export * from './primary';
