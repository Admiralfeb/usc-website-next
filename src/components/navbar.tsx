import { useMediaQuery, useTheme } from '@material-ui/core';
import { navItems } from 'data/navItems';
import { useRouter } from 'next/router';
import { useMemo } from 'react';
import { NavbarMobile } from './navbar.mobile';
import { NavbarFull } from './navbar.full';

const ResponsiveNavbar = ({
  isMobile,
  title,
}: {
  isMobile: boolean;
  title: string;
}) => {
  return isMobile ? (
    <div data-testid="mobile">
      <NavbarMobile title={title} navItems={navItems} />
    </div>
  ) : (
    <div data-testid="full">
      <NavbarFull navItems={navItems} />
    </div>
  );
};

export const Navbar = () => {
  const pathName = useRouter().pathname;
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('md'));

  const title = useMemo(
    () => navItems.find((x) => pathName.startsWith(x.to))?.text,
    [pathName]
  );

  return (
    <>
      <ResponsiveNavbar isMobile={isMobile} title={title} />
    </>
  );
};
