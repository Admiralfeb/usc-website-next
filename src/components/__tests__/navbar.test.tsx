import { cleanup, render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'components/builds/builds/node_modules/react-router-dom';
import { NavbarComponent } from 'components/navbar';
import { SnackbarProvider } from 'notistack';
import * as snackbarhooks from 'notistack';
import * as anonhooks from 'hooks/useLoginAnon';

jest.mock('hooks/useLoginAnon', () => ({
  useLoginAnon: jest.fn(),
}));

let componentBody: RenderResult;

describe('Navbar', () => {
  afterEach(cleanup);

  it(`should match the snapshot`, () => {
    componentBody = render(
      <SnackbarProvider maxSnack={3}>
        <MemoryRouter>
          <NavbarComponent />
        </MemoryRouter>
      </SnackbarProvider>
    );
    const { baseElement } = componentBody;
    expect(baseElement).toMatchSnapshot();
  });

  it('should show a snackbar if loginAnon failed', async () => {
    const enqueueSnackbarMock = jest.fn();
    jest.spyOn(snackbarhooks, 'useSnackbar').mockImplementation(() => ({
      enqueueSnackbar: enqueueSnackbarMock,
      closeSnackbar: jest.fn(),
    }));
    jest.spyOn(anonhooks, 'useLoginAnon').mockImplementation(() => {
      throw new Error('oops');
    });

    componentBody = render(
      <SnackbarProvider maxSnack={3}>
        <MemoryRouter>
          <NavbarComponent />
        </MemoryRouter>
      </SnackbarProvider>
    );

    expect(enqueueSnackbarMock).toHaveBeenCalled();
  });
});
