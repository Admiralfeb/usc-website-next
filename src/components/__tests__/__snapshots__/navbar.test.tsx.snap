// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match the snapshot 1`] = `
<body>
  <div>
    <nav
      class="makeStyles-root-1"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-2"
        href="/home"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Home
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-2"
        href="/about"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          About
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-2"
        href="/information"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Information
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-2"
        href="/builds"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          USC Builds
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <div
        class="makeStyles-filler-6"
      />
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-navLinkJoin-4 MuiButton-containedSecondary"
        href="/join"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Join
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </nav>
  </div>
</body>
`;
